src/main/java/io/cryostat/net/AgentClient.java

```diff
@@ -338,4 +278,174 @@ AgentClient create(URI agentUri) {
                     vertx, gson, httpTimeout, webClient, credentialsManager, agentUri, logger);
         }
     }
+
+    private static class AgentRecordingDescriptor implements IRecordingDescriptor {
+
+        final JsonObject json;
+
+        AgentRecordingDescriptor(JsonObject json) {
+            this.json = json;
+        }
+
+        @Override
+        public IQuantity getDataEndTime() {
+            // TODO
+            return UnitLookup.EPOCH_MS.quantity(-1L);
+        }
+
+        @Override
+        public IQuantity getDataStartTime() {
+            // TODO
+            return UnitLookup.EPOCH_MS.quantity(-1L);
+        }
+
+        @Override
+        public IQuantity getDuration() {
+            return UnitLookup.MILLISECOND.quantity(json.getLong("duration"));
+        }
+
+        @Override
+        public Long getId() {
+            return json.getLong("id");
+        }
+
+        @Override
+        public IQuantity getMaxAge() {
+            return UnitLookup.MILLISECOND.quantity(json.getLong("maxAge"));
+        }
+
+        @Override
+        public IQuantity getMaxSize() {
+            return UnitLookup.BYTE.quantity(json.getLong("maxSize"));
+        }
+
+        @Override
+        public String getName() {
+            return json.getString("name");
+        }
+
+        @Override
+        public ObjectName getObjectName() {
+            // TODO Auto-generated method stub
+            return null;
```

Comment:
Should this TODO be done?

Reply:
I'll just remove the comment. This is here because this class needs to implement this JMC interface so that it fits in with the older JMX-backed implementation that borrowed a bunch of JMC classes and interfaces, but nowhere in Cryostat do we call this method, nor do I think it's on any paths where we call into JMC code. The FlightRecorder V2 (which I believe is any open sourced version of JFR) implementation of this interface in JMC just returns null anyway.